src/script/event/NotificationService.ts

```diff
@@ -169,4 +168,33 @@ export class NotificationService {
       value: notificationId,
     });
   }
+
+  getNotificationIdByMessageId(messageId: string, clientId: string): Promise<string | void> {
+    return new Promise(async resolve => {
+      let message: EventRecord;
+
+      if (this.storageService.isTemporaryAndNonPersistent) {
+        message = Object.values(this.storageService.objectDb.events).filter(event => event.id === messageId)[0];
+      } else {
+        message = await this.storageService.db
+          .table(StorageSchemata.OBJECT_STORE.EVENTS)
+          .where({id: messageId})
+          .first();
+      }
+
+      const {notifications} = await this.getNotifications(clientId);
+
+      notifications.forEach(notification => {
+        notification.payload.forEach(event => {
+          if (event.type === CONVERSATION_EVENT.OTR_MESSAGE_ADD) {
+            const eventLooksLikeMessage =
+              event.time === message.time && event.from === message.from && event.conversation === message.conversation;
+            if (eventLooksLikeMessage) {
+              resolve(notification.id);
```

Comment:
The loop will keep on looping and there is nothing you can do ...
You might want to use a `.find()` here (in place of both instances of `forEach`) or two `for .. of` loops and a return.